### PR TITLE
[PULS-89] Web - message templates

### PR DESCRIPTION
**Pull request by mohsintahir**

### CODE REVIEW

File: src/assets/images/vector/templates-dark.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="#F5F5F5" width="24" height="24">
+    <g><rect fill="none" height="24" width="24"/></g><g><g><g><path d="M2.5,4v3h5v12h3V7h5V4H2.5z M21.5,9h-9v3h3v7h3v-7h3V9z"/></g></g></g>
+</svg>
```
Comment (Serubin) [inline src/assets/images/vector/templates-dark.svg:3]: [nit] Add a new line

File: src/assets/images/vector/templates.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="#666666" width="24" height="24">
+    <g><rect fill="none" height="24" width="24"/></g><g><g><g><path d="M2.5,4v3h5v12h3V7h5V4H2.5z M21.5,9h-9v3h3v7h3v-7h3V9z"/></g></g></g>
+</svg>
```
Comment (Serubin) [inline src/assets/images/vector/templates.svg:3]: [nit] Add a new line

File: src/components/Account/Templates/CreateTemplate.vue [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        height: 100%;
+        margin: 5em auto;
+        .loading-center {
+            position: absolute;
+            top: 50%;
+            left: 50%;
+            transform: translateX(-50%);
+        }
+    }
+</style>
```
Comment (Serubin) [inline src/components/Account/Templates/CreateTemplate.vue:103]: [nit] Add a new line

File: src/components/Account/Templates/EditTemplate.vue [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        height: 100%;
+        margin: 5em auto;
+        .loading-center {
+            position: absolute;
+            top: 50%;
+            left: 50%;
+            transform: translateX(-50%);
+        }
+    }
+</style>
```
Comment (Serubin) [inline src/components/Account/Templates/EditTemplate.vue:103]: [nit] Add a new line

File: src/components/Account/Templates/CreateTemplate.vue [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+        handleCreated() {
+            this.loading = false;
+            this.$router.push({ name: 'templates' });
+        },
+
+        cancel () {
+            this.$router.push({ name: 'templates' });
+        }
+
```
Comment (Serubin) [inline src/components/Account/Templates/CreateTemplate.vue:84]: Extra line

File: src/components/Account/Templates/CreateTemplate.vue [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        createTemplateText() {
+            
+            if(this.templateText == '')
+                return;
+            
+            this.loading = true;
+            
+            Api.templates.create(this.templateText)
+                .then((data) => this.handleCreated(data.data));
+
```
Comment (Serubin) [inline src/components/Account/Templates/CreateTemplate.vue:73]: Extra like

File: src/utils/api/templates.js [hunk 7]
```diff
@@ -28,4 +29,31 @@ export default class Templates {
...
+    static create(text) {
+        let request = {
+            account_id: store.state.account_id,
+            device_id: Api.generateId(),
+            text: Crypto.encrypt(text)
+        };
+
+        let constructed_url = Url.get('create_template');
+
+        const promise = new Promise((resolve) => {
```
Comment (Serubin) [inline src/utils/api/templates.js:42]: You can just return here. No need to store the promise

File: src/utils/api/templates.js [hunk 8]
```diff
@@ -28,4 +29,31 @@ export default class Templates {
...
+            axios.post(constructed_url, request, { 'Content-Type': 'application/json' })
+                .then(response => { resolve(response) });
+        });
+
+        return promise;
+    }
+
+    static update(id, text) {
+        let constructed_url = Url.get('update_template') + id + Url.getAccountParam() + "&text=" + text;
+        const promise = new Promise((resolve) => {
```
Comment (Serubin) [inline src/utils/api/templates.js:52]: Same here - prefer immediate return

**Review by Serubin — CHANGES_REQUESTED** [review]
I've left a bunch of nits. Please lint this code - our style guide was not followed.

File: src/assets/images/vector/templates-dark.svg (hunk 1 above)
Comment (mohsintahir) [reply src/assets/images/vector/templates-dark.svg:3]: Can you kindly tell me that why in other existing svg's, we didn't add new line at the end and why do we need here ?

File: src/utils/api/templates.js (hunk 8 above)
Comment (mohsintahir) [reply src/utils/api/templates.js:52]: Hi, Thank you so much for reviewing the PR. For this suggestion, i think it is personal recommendation of how to return i.e. whether storing in a var or directly.

**Comment by mohsintahir** [comment]
hi @Serubin, thank you for reviewing it. I ran lint and it is not showing me any problems.

**Review by Serubin — CHANGES_REQUESTED** [review]
Please ensure your linting is setup correctly. There are still several issues

**Comment by mohsintahir** [comment]
@Serubin would you guide me in setting up ? could you share some article, tutorial ?

**Comment by Serubin** [comment]
It seems our linting broke

**Comment by mohsintahir** [comment]
@Serubin it seems so.

**Comment by Serubin** [comment]
I made two PRs fixing linting.

**Comment by mohsintahir** [comment]
Thanks @Serubin

**Pull request merged**